pkg/api_keys/postgres.go

```diff
@@ -18,11 +20,28 @@ func NewPostgresRepo(db *gorm.DB) Repository {
 // GetApiKeyByUserId returns the api key for a given user id
 func (r *repo) GetApiKeyByUserId(userId *uuid.UUID) (*uuid.UUID, error) {
 	// TODO: implement this
-	return nil, nil
+	var apiKey uuid.UUID
+	err := r.DB.Where("user_id = ?", userId).First(&apiKey).Error
+	if err != nil {
+		return nil, err
+	}
+
+	return &apiKey, nil
+
+	//return nil, nil
 }
 
 // ApiAuthCheck checks if the api key is valid for the given user id
-func (r *repo) ApiAuthCheck(ApiKey *uuid.UUID) (bool, error) {
+func (r *repo) ApiAuthCheck(ApiKeys *uuid.UUID) (bool, error) {
 	// TODO: implement this
+	var count int64
```

Comment:
Api Key is Primary Key for user, a direct match would have worked.

Reply:
Can we make changes and submit again or this will be final submission?

Reply:
I didn't get you. Can you please elaborate a little ?